source/fundamentals/builders/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1198 @@
+       | `switchBooleanOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlBoolean.html#switchBooleanOn(java.util.function.Function)>`__
+       | `switchDateOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDate.html#switchDateOn(java.util.function.Function)>`__
+       | `switchDocumentOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#switchDocumentOn(java.util.function.Function)>`__
+       | `switchIntegerOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlInteger.html#switchIntegerOn(java.util.function.Function)>`__
+       | `switchMapOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#switchMapOn(java.util.function.Function)>`__
+       | `switchNumberOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlNumber.html#switchNumberOn(java.util.function.Function)>`__
+       | `switchStringOn() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#switchStringOn(java.util.function.Function)>`__
+
+     - :manual:`$switch </reference/operator/aggregation/switch/>`
+
+.. include:: /includes/fundamentals/agg-mql-diff.rst
+
+Suppose you have a collection of customers with their membership information.
+Originally, customers were either members or not. Over time, membership levels
+were introduced and used the same field. The information stored in this field
+can be one of a few different types, and you want to create a standardized value
+indicating their membership level.
+
+The ``switchOn()`` method runs through possible clauses without assuming the
+type of the ``member`` field. Once the ``switchOn()`` method determines the
+data type of the field, it calculates a string value corresponding to the
+membership level. If the data type is a string, it returns the membership
+level value. If the data type is a boolean, it returns either a ``Gold`` or
+``Guest`` membership level. If the data type is an array, it returns the
+most recent membership level, which is the last item in the array. If the
+``member`` field is an unknown type, the ``switchOn()`` method provides a
+default value of ``Guest``.
+
+The following code shows the pipeline for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   asList(project(fields(
+            computed("membershipLevel",
+               current().getField("member").switchOn(field -> field
+                     .isString(s -> s)
+                     .isBoolean(b -> b.cond(of("Gold"), of("Guest")))
+                     .<MqlString>isArray(a -> a.last())
+                     .defaults(d -> of("Guest"))))
+   )));
+
+The following code provides an equivalent aggregation pipeline in
+the Query API:
+
+.. code-block:: javascript
+   :copyable: true
+
+   [
+     {
+       $project: {
+         result: {
```

Comment:
I think this should be `membershipLevel`?

Consider doublechecking the other examples to make sure they match up with the Java.